# Final Testing on New Data

*Author: Grace Campbell*

In [1]:
from models import get_posts, naive_bayes, knn, svc

I have saved the final, "best" iterations of the models for use here. This notebook will scrape /r/TheOnion and /r/News again for new titles so that we can see how the models perform on truly unseen data.

In [2]:
titles = get_posts()

In [3]:
X = titles['title']
y = titles['is_onion']

In [4]:
naive_bayes(X, y)

Accuracy score: 0.9071146245059288
Sensitivity: 0.9453207150368034
Specificity: 0.8430335097001763


In [5]:
knn(X, y)

Accuracy score: 0.7463768115942029
Sensitivity: 1.0
Specificity: 0.32098765432098764


In [6]:
svc(X, y)

Accuracy score: 0.8853754940711462
Sensitivity: 0.9936908517350158
Specificity: 0.7037037037037037


Naive Bayes and SVM hold up quite well to this unseen data, with accuracy scores of 90.7% and 88.5%, respectively. $k$-Nearest Neighbors, however, does not seem to perform as well, with an accuracy score of 74.6% and a specificity of 32%. It seems that the kNN model is simply predicting /r/TheOnion every time. It could be that kNN simply is not the best model for this purpose.